firedrake/mg/utils.py

```diff
@@ -171,3 +166,11 @@ def get_level(obj):
 def has_level(obj):
     """Does the provided object have level info?"""
     return hasattr(obj.topological, "__level_info__")
+
+
+def _cache_key(Vc, Vf):
+    _, levelf = get_level(Vf.mesh())
+    _, levelc = get_level(Vc.mesh())
+    return (entity_dofs_key(Vf.finat_element.entity_dofs())
```

Comment:
@ksagiyam We need `entity_dofs_key` for both spaces, since the elements might differ most times (one will be the coordinate element). The exception is the coarse_cell_to_fine_node_map, where the elements are the same.

Reply:
Fixed.